# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1311719668 -7200
#      Wed Jul 27 00:34:28 2011 +0200
# Node ID 5ef33f99a078223a619bd7e5a00213fb4834f96b
# Parent  ffc2e1d1ad5fd21988ad8cdfe880bb6d0c5bebdd
Command get inserted by double clicking in the command history again.

diff --git a/gui/src/HistoryDockWidget.cpp b/gui/src/HistoryDockWidget.cpp
--- a/gui/src/HistoryDockWidget.cpp
+++ b/gui/src/HistoryDockWidget.cpp
@@ -41,23 +41,30 @@ HistoryDockWidget::construct ()
 
   layout->addWidget (m_historyListView);
   layout->addWidget (m_filterLineEdit);
   layout->setMargin (2);
 
   widget ()->setLayout (layout);
 
   connect (m_filterLineEdit, SIGNAL (textEdited (QString)), this, SLOT (setFilter (QString)));
+  connect (m_historyListView, SIGNAL (doubleClicked (QModelIndex)), this, SLOT (handleDoubleClick (QModelIndex)));
 }
 
 void
 HistoryDockWidget::noticeSettings ()
 {
 
 }
 
 void
-HistoryDockWidget::setFilter(QString filter)
+HistoryDockWidget::setFilter (QString filter)
 {
   m_historyListView->setEnabled (false);
   m_sortFilterProxyModel.setFilterWildcard ( QString ("*%1*").arg (filter));
   m_historyListView->setEnabled (true);
 }
+
+void
+HistoryDockWidget::handleDoubleClick (QModelIndex modelIndex)
+{
+  emit commandDoubleClicked (modelIndex.data().toString());
+}
diff --git a/gui/src/HistoryDockWidget.h b/gui/src/HistoryDockWidget.h
--- a/gui/src/HistoryDockWidget.h
+++ b/gui/src/HistoryDockWidget.h
@@ -32,18 +32,23 @@ Q_OBJECT
 public:
   HistoryDockWidget (QWidget *parent = 0);
   void updateHistory (QStringList history);
 
 public slots:
   /** Tells the widget to notice settings that are probably new. */
   void noticeSettings ();
   void setFilter (QString filter);
+
 signals:
   void information (QString message);
+  void commandDoubleClicked (QString command);
+
+private slots:
+  void handleDoubleClick (QModelIndex modelIndex);
 
 private:
   void construct ();
   QListView *m_historyListView;
   QLineEdit *m_filterLineEdit;
   QSortFilterProxyModel m_sortFilterProxyModel;
 };
 
diff --git a/gui/src/MainWindow.cpp b/gui/src/MainWindow.cpp
--- a/gui/src/MainWindow.cpp
+++ b/gui/src/MainWindow.cpp
@@ -279,16 +279,17 @@ MainWindow::construct ()
   //connect (m_filesDockWidget, SIGNAL (visibilityChanged (bool)), showFileBrowserAction, SLOT (setChecked (bool)));
 
   connect (this, SIGNAL (settingsChanged ()), m_variablesDockWidget, SLOT (noticeSettings ()));
   connect (this, SIGNAL (settingsChanged ()), m_historyDockWidget, SLOT (noticeSettings ()));
   connect (this, SIGNAL (settingsChanged ()), m_filesDockWidget, SLOT (noticeSettings ()));
 
   connect (m_filesDockWidget, SIGNAL (openFile (QString)), this, SLOT (handleOpenFileRequest (QString)));
   connect (m_historyDockWidget, SIGNAL (information (QString)), this, SLOT (reportStatusMessage (QString)));
+  connect (m_historyDockWidget, SIGNAL (commandDoubleClicked (QString)), this, SLOT (handleCommandDoubleClicked (QString)));
   connect (saveWorkspaceAction, SIGNAL (triggered ()), this, SLOT (handleSaveWorkspaceRequest ()));
   connect (loadWorkspaceAction, SIGNAL (triggered ()), this, SLOT (handleLoadWorkspaceRequest ()));
   connect (clearWorkspaceAction, SIGNAL (triggered ()), this, SLOT (handleClearWorkspaceRequest ()));
 
   setWindowTitle (QString (VERSION_STRING));
 
   setCentralWidget (m_centralMdiArea);
   addDockWidget (Qt::LeftDockWidgetArea, m_variablesDockWidget);
